src/action.js

```diff
@@ -27,122 +46,67 @@ class Action extends Emitter {
     this.children = []
 
     if (status) {
-      console.assert(this[status], 'Unexpected task status ' + status)
-      this[status]()
+      this.setState(status)
     }
   }
 
-  get type() {
-    return this.command[this.status]
+  get type(): string {
+    return this.command[this.status] || ''
   }
 
-  /**
-   * Open state. The action has started, but has received no response.
-   * @param {*} [nextPayload]
-   */
-  get open() {
-    return createActionUpdater(this, 'open', false)
+  get open(): ActionUpdater {
+    return this.setState.bind(this, 'open')
```

Comment:
It turns out, `.bind` is faster than returning a new function every time. Also it made things cleaner.